source_middleware.go

```diff
@@ -14,17 +14,42 @@
 
 package sdk
 
+import (
+	"context"
+	"fmt"
+	"strconv"
+	"sync"
+
+	"github.com/conduitio/conduit-commons/config"
+	"github.com/conduitio/conduit-commons/opencdc"
+	"github.com/conduitio/conduit-commons/rabin"
+	"github.com/conduitio/conduit-commons/schema"
+	"github.com/conduitio/conduit-connector-sdk/internal"
+	sdkschema "github.com/conduitio/conduit-connector-sdk/schema"
+)
+
 // SourceMiddleware wraps a Source and adds functionality to it.
 type SourceMiddleware interface {
 	Wrap(Source) Source
 }
 
+// SourceMiddlewareOption can be used to change the behavior of the default source
+// middleware created with DefaultSourceMiddleware.
+type SourceMiddlewareOption func(SourceMiddleware)
+
 // DefaultSourceMiddleware returns a slice of middleware that should be added to
 // all sources unless there's a good reason not to.
-func DefaultSourceMiddleware() []SourceMiddleware {
-	return []SourceMiddleware{
-		// TODO add default middleware (e.g. rate limit)
+func DefaultSourceMiddleware(opts ...SourceMiddlewareOption) []SourceMiddleware {
+	middleware := []SourceMiddleware{
+		&SourceWithSchema{},
```

Comment:
WDYT about assigning default values here? I think it would make it easier to discover it (since at least I would visit this method first).

Reply:
That's what I had before, but then I changed it, otherwise the defaults would be hardcoded here. If someone wanted to use this middleware with the same default configuration, they would have to copy over this part, and if we change the defaults, the copied code won't reflect the changes. So why I prefer this approach, is that the empty value already reflects the default behavior.